it/controllers/benefits/MileageBenefitAmountControllerISpec.scala

```diff
@@ -0,0 +1,513 @@
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel.copy(mileage = None))))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlGet(url(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          import Selectors._
+          import user.commonExpectedResults._
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption(taxYearEOY))
+          buttonCheck(user.commonExpectedResults.continueButtonText, continueButtonSelector)
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraph, contentSelector)
+          textOnPageCheck(hintText, hintTestSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("", amountField)
+          formPostLinkCheck(continueLink, formSelector)
+          welshToggleCheck(user.isWelsh)
+
+        }
+
+        "render the mileage amount page with amount pre-filled" which {
+
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel)))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlGet(url(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          import Selectors._
+          import user.commonExpectedResults._
+
+          s"has an OK($OK) status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption(taxYearEOY))
+          buttonCheck(user.commonExpectedResults.continueButtonText, continueButtonSelector)
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraphWithPrefill, contentSelector)
+          textOnPageCheck(hintText, hintTestSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("400", amountField)
+          formPostLinkCheck(continueLink, formSelector)
+          welshToggleCheck(user.isWelsh)
+
+        }
+
+        "redirect" when {
+
+          "mileage question is no" which {
```

Comment:
this goes different places if its prior or not right? and mileage question = no, so do you need another test?